core/block/editor/template.go

```diff
@@ -56,27 +51,19 @@ func (t *Template) CreationStateMigration(ctx *smartblock.InitContext) migration
 	})
 }
 
-// GetNewPageState returns state that can be safely used to create the new document
-// it has not localDetails set
-func (t *Template) GetNewPageState(name string) (st *state.State, err error) {
-	st = t.NewState().Copy()
-
-	if err = t.updateTypeKey(st); err != nil {
-		return nil, err
-	}
-
-	st.RemoveDetail(bundle.RelationKeyTargetObjectType.String(), bundle.RelationKeyTemplateIsBundled.String())
-	st.SetDetailAndBundledRelation(bundle.RelationKeySourceObject, pbtypes.String(t.Id()))
-	// clean-up local details from the template state
-	st.SetLocalDetails(nil)
-
-	if name != "" {
-		st.SetDetail(bundle.RelationKeyName.String(), pbtypes.String(name))
-		if title := st.Get(template.TitleBlockId); title != nil {
-			title.Model().GetText().Text = ""
+func (t *Template) UpdateTypeKey(st *state.State) error {
```

Comment:
move to service too

Reply:
Done! Moved to service as private method, however it has grown up in cause of error checks:
https://github.com/anyproto/anytype-heart/pull/570/files#diff-4985ac9c720e3e9eacb640d7fa1e414316d44caef810683fd193633ad5c4a018R283

Reply:
Maybe we can get rid of returning any error at all?
1. First we try to get typeKey from store by targetObjectType that are set to template's details
2. If we fail we will try to get it from ObjectTypeKeys of template's state. We expect it to have two values - template and the target one. However, this slice could be invalid as well, e.g. values contain "ot-"
3. In case we fail to gather typeKey from ObjectTypeKeys we can leave state as is or fail whole ObjectCreate/ObjectApplyTemplate command

Reply:
@deff7 , how do you think, should we modify this function according logic described above?